velox/core/CancelPool.h

```diff
@@ -167,6 +167,7 @@ class CancelPool {
   }
 
   void requestTerminate() {
+    std::lock_guard<std::mutex> l(mutex_);
```

Comment:
I see that some methods of CancelPool are still not safe, namely:

shouldStop()
terminateRequested() - this is not used from what I can see, so we should probably delete it.
pauseRequested()